txscript/engine.go

```diff
@@ -285,6 +285,33 @@ type Engine struct {
 	witnessProgram  []byte
 	inputAmount     int64
 	taprootCtx      *taprootExecutionCtx
+
+	// StepCallback is an optional function that will be called every time
+	// a step has been performed during script execution.
+	//
+	// NOTE: This is only meant to be used in debugging, and SHOULD NOT BE
+	// USED during regular operation.
+	StepCallback func(*StepInfo) error
```

Comment:
Can be a private variable? In that maybe we introduce a new specific constructor? Like `txscript.NewDebugEngine`.

Reply:
Fixed: https://github.com/btcsuite/btcd/pull/1980/commits/55a376e6ea591ba5a6e3ed2302787c95c8135d06